chrome/content/zotero/xpcom/itemPaneManager.js

```diff
@@ -0,0 +1,335 @@
+ * @property {(l10nArgs: string) => void} setL10nArgs - Set l10n args for section header
+ * @property {(l10nArgs: string) => void} setEnabled - Set pane enabled state
+ * @property {(summary: string) => void} setSectionSummary - Set pane section summary
+ * @property {(buttonType: string, options: {disabled?: boolean; hidden?: boolean}) => void} setSectionButtonStatus - Set pane section button status
+ * @typedef SectionHookArgs
+ * @type {SectionBasicHookArgs & SectionUIHookArgs}
+ * @typedef  {SectionHookArgs & { refresh: () => Promise<void> }} SectionInitHookArgs
+ * A `refresh` is exposed to plugins to allows plugins to refresh the section when necessary,
+ * e.g. item modify notifier callback. Note that calling `refresh` during initialization
+ * have no effect.
+ * @typedef  {SectionHookArgs & { incomingData: { type: "item" | "mode" | "inTrash" | "tabType", value: any } }} SectionDataChangeHookArgs
+ * @typedef  {SectionHookArgs & { event: Event }} SectionEventHookArgs
+ * @typedef ItemDetailsSectionOptions
+ * @type {object}
+ * @property {string} paneID - Unique pane ID
+ * @property {string} pluginID - Set plugin ID to auto remove section when plugin is disabled/removed
+ * @property {SectionL10n & SectionIcon} head - Header options. Icon should be 16*16 and `label` need to be localized
+ * @property {SectionL10n & SectionIcon} sidenav - Sidenav options. Icon should be 20*20 and `tooltiptext` need to be localized
+ * @property {string} [fragment] - Pane fragment as string
+ * @property {(options: SectionInitHookArgs) => void} [onInit]
+ * Lifecycle hook called when section is initialized
+ *
+ * Do:
+ * 1. Initialize data if necessary
+ * 2. Setup hooks, e.g. notifier callback
+ *
+ * NOT do:
+ * 1. Render/refresh UI
+ * @property {(options: SectionBasicHookArgs) => void} [onDestroy]
+ * Lifecycle hook called when section is destroyed
+ *
+ * Do:
+ * 1. Remove data and release resource
+ * 2. Remove hooks, e.g. notifier callback
+ *
+ * NOT do:
+ * 1. Render/refresh UI
+ * @property {(options: SectionDataChangeHookArgs) => boolean} [onDataChange]
+ * Lifecycle hook called when section incoming data change received
+ *
+ * Return `false` to abort the data change, otherwise the data change will be applied
+ * (stored internally and can be accessed by `options.getData` in other hooks).
+ *
+ * Do:
+ * 1. Validate and decide whether to accept the incoming data change
+ * 2. Update the section enabled state with `options.setEnabled`. For example, if the section
+ *   is only enabled in the readers, you can use:
+ * ```js
+ * onDataChange(options) {
+ *   let { incomingData, setEnabled } = options.;
+ *   if (incomingData.type === "tabType") {
+ *     setEnabled(incomingData.value === "reader");
+ *   }
+ *   return true;
+ * }
+ * ```
+ * options.setEnabled(options.tabType === "reader");
+ *
+ * NOT do:
+ * 1. Render/refresh UI
+ * @property {(options: SectionHookArgs) => void | Promise<void>} onRender
```

Comment:
`options` seems a little wrong for these handlers. This is mostly a documentation issue, since people can name the argument whatever they want (or just use destructing assignment), but while `options` is totally appropriate for passing things like ids and strings and event handlers when registering a section, here things are just getting passed to the handlers regardless, so they're not options. Maybe just `props`, and encourage destructuring assignment in the documentation?